core/local/analysis.js

```diff
@@ -38,37 +38,55 @@ const panic = (context, description) => {
   throw new Error(description)
 }
 
-function analyseEvents (events /*: LocalEvent[] */, pendingChanges /*: LocalChange[] */) /*: LocalChange[] */ {
-  const stopMeasure = measureTime('LocalWatcher#analyseEvents')
-  // OPTIMIZE: new Array(events.length)
-  const changes /*: LocalChange[] */ = []
-  const changesByInode /*: Map<number, LocalChange> */ = new Map()
-  const changesByPath /*: Map<string, LocalChange> */ = new Map()
-  const getChangeByInode = (e) => {
-    const ino = getInode(e)
-    if (ino) return changesByInode.get(ino)
+class LocalChangeMap {
+  /*::
+  changes: LocalChange[]
+  changesByInode: Map<number, LocalChange>
+  changesByPath: Map<string, LocalChange>
+  */
+
+  constructor () {
+    this._clear()
+  }
+
+  _clear () {
+    this.changes = []
+    this.changesByInode = new Map()
+    this.changesByPath = new Map()
+  }
+
+  findByInode (ino /*: ?number */) /*: ?LocalChange */ {
+    if (ino) return this.changesByInode.get(ino)
     else return null
   }
-  const withChangeByPath = (e, callback) => {
-    return callback(changesByPath.get(e.path))
+
+  whenFoundByPath /*:: <T> */ (path /*: string */, callback /*: (LocalChange) => T */) /*: ?T */ {
+    const change = this.changesByPath.get(path)
+    if (change) return callback(change)
   }
-  const changeFound = (c /*: ?LocalChange | true */) => {
-    if (c == null) {
-      // No change was found. Nothing to do.
-      return
-    }
-    if (c === true) {
-      // A previous change was transformed. Nothing more to index.
-      return
-    }
-    changesByPath.set(c.path, c)
-    if (typeof c.ino === 'number') changesByInode.set(c.ino, c)
-    else changes.push(c)
+
+  put (c /*: LocalChange */) {
+    this.changesByPath.set(c.path, c)
+    if (typeof c.ino === 'number') this.changesByInode.set(c.ino, c)
+    else this.changes.push(c)
+  }
```

Comment:
I didn't understand the code at first reading. Because I could not understand what it did.
It put the same value in different objects indexed in different ways.